### PR TITLE
CL F4 target.  Motor stop causes throttle to not respond above idle

### PR DESCRIPTION
**Pull request by hattrickmwf**
On 3.2 RC 4 CL F4 board.  I set up the quad with the minimum amount of stuff to make it flyable.  Airmode on a switch and with motor stop on.  No throttle.  Motors will spin when armed and airmode turned on, but no higher than idle.  Turned motor stop off.  Still no throttle.  Reflashed and set up the minimum with airmode as a feature and motor stop off.  Throttle works!  Made the rest of the changes I normally make.  Throttle not working again.  Reflashed,  set up the minimum required, airmode as a feature and motor stop off,  throttle works again.  Made the changes I normally make one at a time and checked if throttle worked after each one and it stayed working with every change.  Very strange.  It seems that making more than one change at a time somehow made the throttle stop working.  While making them one at a time kept the throttle working.
All works as normal on 3.17

### CODE REVIEW

**Comment by mikeller** [comment]
@hattrickmwf: I think what you wanted to do is to open an issue. What you have done is opening a pull request, with no code attached to it. Please close this and open an issue here: https://github.com/cleanflight/cleanflight/issues

File: src/main/common/crc.c [hunk 1]
```diff
@@ -77,3 +77,25 @@ void crc8_dvb_s2_sbuf_append(sbuf_t *dst, uint8_t *start)
...
+    const uint8_t *p = (const uint8_t *)data;
+    const uint8_t *pend = p + length;
+
+    for (; p != pend; p++) {
+        crc ^= *p;
+    }
+    return crc;
+}
+
+void crc8_xor_sbuf_append(sbuf_t *dst, uint8_t *start)
```
Comment (ledvinap) [inline src/main/common/crc.c]: (I'm not a big fan of this kind of functions, but that's mostly personal feeling) ...

File: src/main/common/crc.c [hunk 2]
```diff
@@ -77,3 +77,25 @@ void crc8_dvb_s2_sbuf_append(sbuf_t *dst, uint8_t *start)
...
+{
+    uint8_t crc = 0;
+    const uint8_t *end = dst->ptr;
+    for (uint8_t *ptr = start; ptr < end; ++ptr) {
```
Comment (ledvinap) [inline src/main/common/crc.c]: Calling `crc8_xor_update` would reduce code duplication ...

File: src/main/common/streambuf.c [hunk 3]
```diff
@@ -20,6 +20,13 @@
 
 #include "streambuf.h"
 
+sbuf_t *sbufInit(sbuf_t *sbuf, uint8_t *ptr, uint8_t *end)
```
Comment (ledvinap) [inline src/main/common/streambuf.c]: in most cases `uint8_t *ptr, int len` would be easier to use. The `end` is internal optimization of `sbuf`

File: src/main/common/crc.c (hunk 2 above)
Comment (martinbudden) [reply src/main/common/crc.c]: OK

File: src/main/common/streambuf.c (hunk 3 above)
Comment (martinbudden) [reply src/main/common/streambuf.c]: I'll need to check. I thought this was probably more common, but not sure.

**Comment by hattrickmwf** [comment]
Thank you.  New to github.  Really don’t know what I’m doing!  

 

Mike Fredricks

**Pull request closed without merging**